tools/eslint-rules/prefer-primordials.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @fileoverview We shouldn't use global built-in object for security and
+ *               performance reason. This linter rule reports replacable codes
+ *               that can be replaced with primordials.
+ * @author Leko <leko.noor@gmail.com>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function toPrimordialsName(obj, prop) {
+  return obj + toUcFirst(prop);
+}
+
+function toUcFirst(str) {
+  return str[0].toUpperCase() + str.slice(1);
+}
+
+function isTarget(map, varName) {
+  return map.has(varName);
+}
+
+function isIgnored(map, varName, propName) {
+  if (!map.has(varName) || !map.get(varName).has(propName)) {
+    return false;
+  }
+  return map.get(varName).get(propName).ignored;
+}
+
+function getReportName({ name, parentName, into }) {
+  if (into) {
+    return toPrimordialsName(into, name);
+  }
+  if (parentName) {
+    return toPrimordialsName(parentName, name);
+  }
+  return name;
+}
+
+/**
+ * Get identifier of object spread assignment
+ *
+ * code: 'const { ownKeys } = Reflect;'
+ * argument: 'ownKeys'
+ * return: 'Reflect'
```

Comment:
Is this the best place to have this comment?

JSDoc tags (`@param`, `@returns`, etc.) usually go here so I'm a little bit confused about what this means.

Reply:
@DerekNonGeneric I'm going to rewrite the comment as below. How about it?

```
/**
 * Get identifier of object spread assignment.
 *
 * For example, suppose you run linter on the following code. Then you specify
 * Identifier of `ownKeys` as an argument, this will return `'Reflect'`.
 *
 * const { ownKeys } = Reflect;
 */
```

Reply:
Unfortunately, my current understanding of this code is zip, so I can't really say&hellip; lol

That certainly looks a lot better to me, though. The more comments the better IMO, but that's just me.